packages/number-field/src/NumberField.ts

```diff
@@ -495,7 +495,17 @@ export class NumberField extends TextfieldBase {
         // Step shouldn't validate when 0...
         if (this.step) {
             const min = typeof this.min !== 'undefined' ? this.min : 0;
-            const moduloStep = (value - min) % this.step;
+            const digitsAfterDecimals =
+                this.step != Math.floor(this.step)
+                    ? this.step.toString().split('.')[1].length
```

Comment:
If `this.step` is not a floating-point number, calling split('.')[1] on it will result in an error. Check first if `this.step` is a floating point number or not.

Reply:
I am using a ternary operator where if `this.step` is not a floating-point number, then ` this.step != Math.floor(this.step)` will evaluate to `false` and hence `digitsAfterDecimals` will be assigned `0`. (Line 501)